mobilecoind/src/monitor_store.rs

```diff
@@ -63,31 +76,117 @@ impl MonitorData {
             ));
         }
 
-        Ok(Self {
-            account_key,
-            first_subaddress,
-            num_subaddresses,
-            first_block,
-            // The next block we need to sync is our first block.
-            next_block: first_block,
-            name: name.to_owned(),
-        })
+        if let Some(pw) = password_hash {
+            let (key, nonce) = Self::expand_password_hash(&pw)?;
+
+            let key_arr = GenericArray::from_slice(&key[..]);
+            let nonce_arr = GenericArray::from_slice(&nonce[..]);
+
+            // Get cipher from hash bytes
+            let cipher = Aes256Gcm::new(key_arr);
+
+            let plaintext_bytes = mc_util_serial::encode(&account_key);
+
+            let ciphertext = cipher
+                .encrypt(nonce_arr, &plaintext_bytes[..])
+                .expect("encryption failure!"); // NOTE: handle this error to avoid panics!
```

Comment:
iirc, the aes-gcm encrypt function only panics if the plaintext bytes are like > 4GB or something

Reply:
you could return the AEAD error if you want though